### PR TITLE
Support arguments from ext

### LINKED ISSUE #56: Add docs re: PHP Extensions required
The agent requires at least require the Socket extension of PHP to run. We should make that list explicit.

### PR DESCRIPTION
**Pull request by asgrim**
https://trello.com/c/QV1zmJfi/26-record-arguments-from-logged-calls-eg-url-for-filegetcontents

Also fixes #56

### CODE REVIEW

File: tests/Integration/AgentTest.php [hunk 1]
```diff
@@ -129,20 +132,23 @@ public function testLoggingIsSent() : void
...
                         $fileGetContentsSpanId = $this->assertUnserializedCommandContainsPayload('StartSpan', ['operation' => 'file_get_contents', 'parent_id' => $controllerSpanId], next($commands), 'span_id');
+                        $this->assertUnserializedCommandContainsPayload('TagSpan', ['span_id' => $fileGetContentsSpanId, 'tag' => 'arguments', 'value' => [__FILE__]], next($commands), null);
                         $this->assertUnserializedCommandContainsPayload('StopSpan', ['span_id' => $fileGetContentsSpanId], next($commands), null);
                     }
 
                     $fooSpanId = $this->assertUnserializedCommandContainsPayload('StartSpan', ['operation' => 'Test/foo'], next($commands), 'span_id');
 
                     if (TestHelper::scoutApmExtensionAvailable()) {
                         $fileGetContentsSpanId = $this->assertUnserializedCommandContainsPayload('StartSpan', ['operation' => 'file_get_contents', 'parent_id' => $fooSpanId], next($commands), 'span_id');
+                        $this->assertUnserializedCommandContainsPayload('TagSpan', ['span_id' => $fileGetContentsSpanId, 'tag' => 'arguments', 'value' => [__FILE__]], next($commands), null);
```
Comment (asgrim) [inline tests/Integration/AgentTest.php]: @cschneid  Are we happy with representing the arguments in this way? or is there another mechanism for doing something like this?

File: src/Agent.php [hunk 2]
```diff
@@ -184,7 +184,8 @@ private function addSpansFromExtension() : void
         }
 
         foreach ($this->phpExtension->getCalls() as $recordedCall) {
-            $this->request->startSpan($recordedCall->functionName(), $recordedCall->timeEntered());
+            $callSpan = $this->request->startSpan($recordedCall->functionName(), $recordedCall->timeEntered());
+            $callSpan->tag('arguments', $recordedCall->arguments());
```
Comment (cschneid) [inline src/Agent.php]: This is dangerous, we'll leak secrets and PII if they are passed in an instrumented call.

We need a whitelist of data we grab, not just all args I think.

Comment (cschneid) [reply src/Agent.php]: Also for historical reasons, this should be tagged `desc` when the data ends up as a tag. That's the magic "anything goes" field that shows up in our UI.

Comment (asgrim) [reply src/Agent.php]: > This is dangerous, we'll leak secrets and PII if they are passed in an instrumented call.

A very good point - would we not run that risk anyway potentially, even if we just grab the `$url` parameter for `file_get_contents` for example?

And I'll update to be tagged as `desc` :+1:

**Comment by asgrim** [comment]
Note to self: also need to add `ext-scoutapm` as a `suggests` in `composer.json`

**Pull request merged**